torch/jit/__init__.py

```diff
@@ -654,7 +654,7 @@ def trace(func,
         return func
     executor_options = {'optimize': bool(optimize)}
     # Special case for common case of passing a single Tensor
-    if isinstance(example_inputs, torch.Tensor):
+    if isinstance(example_inputs, (torch.Tensor, dict)):
```

Comment:
what's this for?

Reply:
Since, like tuples, dicts are iterable, we have to do this, otherwise, we'll call `forward` with a flattened key in each arg slot.

It's actually an interesting question for lists what we should choose. I guess I will have to let top-level list inputs flatten still, to allow backwards compat, since it would be a breaking change otherwise.